akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -367,7 +367,8 @@ object ShardRegion {
    *
    * If gathering the shard information times out the set of shards will be empty.
    */
-  @SerialVersionUID(1L) final case class CurrentShardRegionState(shards: Set[ShardState]) {
+  @SerialVersionUID(1L) final case class CurrentShardRegionState(val shards: Set[ShardState], val failed: Set[ShardId])
```

Comment:
Because this has no bin compat issue, having not been serialized before, I elected to not convert to a class to keep it simple.